Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -679,5 +694,23 @@ TEST_F(SerializerTest, EncodesNonEmptyDocument) {
   ExpectRoundTrip(key, fields, update_time, proto);
 }
 
+TEST_F(SerializerTest, DecodesNoDocument) {
+  // We can't actually *encode* a NoDocument; the method exposed by the
+  // serializer requires both the document key and contents (as an ObjectValue,
+  // i.e. map.) The contents can be empty, but not missing.  As a result, this
+  // test will only verify the ability to decode a NoDocument.
+
+  DocumentKey key = DocumentKey::FromPathString("path/to/the/doc");
+  SnapshotVersion read_time = SnapshotVersion{{1234, 5678}};
+
+  v1beta1::BatchGetDocumentsResponse proto;
+  proto.set_missing(serializer.EncodeKey(key));
+  google::protobuf::Timestamp* read_time_proto = proto.mutable_read_time();
+  read_time_proto->set_seconds(1234);
```

Comment:
Nit: this duplicates the constant used when creating `read_time`, so perhaps it could be
```cpp
read_time_proto->set_seconds(read_time.seconds());
// Same for nanoseconds
```
That's presuming they are supposed to be the same.

Reply:
Yeah, they're all supposed to be the same. Done. (Also added named parameter comments to the SnapshotVersion ctor to make it clearer that these are seconds/nanos.)